src/performance/dashboard/src/components/modals/__tests__/ModalModifyLoadTests.test.js

```diff
@@ -0,0 +1,207 @@
+        const textField = getByPlaceholderText(/Concurrent threads/i);
+        fireEvent.input(textField, { target: { value: sampleFieldValue } });
+        expect(textField.value).toEqual(sampleFieldValue);
+        expect(textField.getAttribute("aria-invalid")).toBe("true");
+        expect(document.querySelector('#concurrency-error-msg').innerHTML).toBe('Must be an integer between 1 and 100')
+    })
+
+
+    // Test duration Field
+    ////////////////////////////////////////////
+    test('the dialog has a duration field that accepts a valid input', () => {
+        const sampleFieldValue = '30';
+        const { getByPlaceholderText } = render(wrapper);
+        const textField = getByPlaceholderText(/Test run duration/i);
+        fireEvent.input(textField, { target: { value: sampleFieldValue } });
+        expect(textField.value).toEqual(sampleFieldValue);
+        expect(textField.getAttribute("aria-invalid")).toBeNull();
+        expect(document.querySelector('#maxSeconds-error-msg')).toBeNull();
+    })
+
+    test('the dialog has a duration field rejects invalid input', () => {
+        const sampleFieldValue = '0';
+        const { getByPlaceholderText } = render(wrapper);
+        const textField = getByPlaceholderText(/Test run duration/i);
+        fireEvent.input(textField, { target: { value: sampleFieldValue } });
+        expect(textField.value).toEqual(sampleFieldValue);
+        expect(textField.getAttribute("aria-invalid")).toBe("true");
+        expect(document.querySelector('#maxSeconds-error-msg').innerHTML).toBe('Must be an integer between 10 and 500')
+    })
+
+    // Test JSON BODY
+    ////////////////////////////////////////////
+    test('the dialog has a body field that accepts a valid json', () => {
+        const sampleFieldValue = '{"field1":"test1", "field2":"test2"}';
+        const { getByPlaceholderText } = render(wrapper);
+        const textField = getByPlaceholderText(/{"id": 1, "message":"hello"}/i);
+        fireEvent.input(textField, { target: { value: sampleFieldValue } });
+        expect(textField.value).toEqual(sampleFieldValue);
+        expect(textField.getAttribute("aria-invalid")).toBeNull();
+        expect(document.querySelector('#body-error-msg')).toBeNull();
+    })
+
+    test('the dialog has a body field that rejects invalid json', () => {
+        const sampleFieldValue = '{"field1"::"test1", "field2":"test2"}'; // invalid JSON ::
+        const { getByPlaceholderText } = render(wrapper);
+        const textField = getByPlaceholderText(/{"id": 1, "message":"hello"}/i);
+        fireEvent.input(textField, { target: { value: sampleFieldValue } });
+        expect(textField.value).toEqual(sampleFieldValue);
+        expect(textField.getAttribute("aria-invalid")).toBe("true");
+        expect(document.querySelector('#body-error-msg').innerHTML).toBe('SyntaxError: Unexpected token : in JSON at position 10')
+    })
+
+
+    // Check form is complete and valid
```

Comment:
Could there maybe be some detail on the form? Assume this is entire form filled when loadrun settings are modified, and the tests check the buttons become active and inactive correctly, just not sure the comment is helpful

Reply:
Removed comment - included in output